src/frontend/src/catalog/system_catalog/rw_catalog/iceberg_tables.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::{Context, anyhow};
+use clap::ValueEnum;
+use risingwave_common::acl::AclMode;
+use risingwave_common::bail;
+use risingwave_common::catalog::{Field, Schema};
+use risingwave_common::config::MetaBackend;
+use risingwave_common::types::DataType::Varchar;
+use risingwave_common::types::Fields;
+use risingwave_connector::parser::postgres_row_to_owned_row;
+use risingwave_frontend_macro::system_catalog;
+use risingwave_pb::user::grant_privilege::Object as GrantObject;
+use thiserror_ext::AsReport;
+
+use crate::catalog::root_catalog::SchemaPath;
+use crate::catalog::system_catalog::SysCatalogReaderImpl;
+use crate::error::{ErrorCode, Result};
+
+// JDBC/SQL catalog integration docs: https://iceberg.apache.org/docs/latest/jdbc/#configurations
+// `iceberg_tables` definition in iceberg java sdk https://github.com/apache/iceberg/blob/4850b622c778deb4b234880bfd7643070e0a5458/core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java#L125-L146
+// This system table is used to store the iceberg tables' metadata and only show the tables that the user has access to,
+// so it can be used by other query engine to fetch iceberg catalog and provide an access control layer.
+
+#[derive(Fields)]
+#[primary_key(catalog_name, table_namespace, table_name)]
+struct IcebergTables {
+    pub catalog_name: String,
+    pub table_namespace: String,
+    pub table_name: String,
+    pub metadata_location: Option<String>,
+    pub previous_metadata_location: Option<String>,
+    pub iceberg_type: Option<String>,
+}
+
+#[system_catalog(table, "rw_catalog.iceberg_tables")]
```

Comment:
`iceberg_tables` sound a bit general, but it's only for hosted catalog (unlike `rw_iceberg_files` which is for all sources).

What about using a new system catalog (e.g., `hosted_iceberg_catalog`?)

Reply:
For JDBC, not every database has schema, so we need to create `iceberg_tables` at the default schema level so that, other query engines can access this view. Therefore, we should keep it in `rw_catalog`.

Reply:
Why can't we make `hosted_iceberg_catalog` also at the default schema level?

Reply:
Oh, since users will be able to select without schema directly, so there's no too much difference whether it's in `rw_catalog` or  `hosted_iceberg_catalog ` 🤣

Reply:
Just confirm with @yezizp2012 that we need to add a migration script to add the newly created schema to all existing created databases, which seems a bit overkill.